projects/lights_control/inc/lights_gpio.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include "event_queue.h"
+
+#include "status.h"
+
+typedef enum { LIGHTS_BOARD_FRONT = 0, LIGHTS_BOARD_REAR, NUM_LIGHTS_BOARDS } LightsBoard;
+
+typedef enum {
+  LIGHTS_GPIO_INIT_MODE_NORMAL = 0,
+  LIGHTS_GPIO_INIT_MODE_TEST_FRONT,
+  LIGHTS_GPIO_INIT_MODE_TEST_REAR,
+  NUM_LIGHTS_GPIO_INIT_MODES
+} LightsGPIOInitMode;
+
+// initializes all the GPIO peripherals
+StatusCode lights_gpio_init();
+
+// used by other modules to get board type
+StatusCode lights_gpio_get_lights_board(LightsBoard *board);
+
+// sets the state of every peripheral related to a specific event
+StatusCode lights_gpio_set(const Event *e);
+
+// returns a pointer to the event mapping tables
+uint16_t *test_lights_gpio_event_mappings(LightsBoard board);
```

Comment:
Why are you exposing a test function in this header? Is this function necessary for testing?

Reply:
yeah because I check the state of the corresponding addresses after setting them to make assertions, so this'll make it easier if I have the mappings.

Reply:
rather than creating the mappings again... that way if we change some mapping I'll have to change the corresponding test as well...

Reply:
Why don't you allow the module to accept a bitset instead of storing the mapping internally or accept the mapping as part of configuration? Then you could store the mapping in a config blob instead and use a simplified test mapping for your tests.

Reply:
@tituschow by simplified mapping you mean like a subset of this mapping? that wouldn't include all the cases though would it?

Reply:
Think about what you're testing - the goal is to verify that your module works. Why would a simplified test mapping not cover all the cases?

Reply:
Well by module working if we mean that the module turns on the correct `gpio` pins for the events given, then the more events I test for the better...  A simplified version of mappings wouldn't contain the mapping for all the events. 
But on the other hand let's say if there's something wrong with the mapping, but since I'm using the same mapping both in my tests and my module, even if the mapping is pointing to the wrong gpio addresses, my tests would still pass which would be a problem. So I do see a point in having a simplified version of mappings in the tests that gets used for testing. 
Correct me if I'm wrong, I may have misunderstood your point.

Reply:
Why would the number of events you're testing matter? More importantly, why would your event even matter? It should work the same if you created some arbitrary test-only set of events and IOs independent of your mapping.

The purpose of the test is not to verify that your mapping is correct, it's to test that your module is correct. Verification of your mapping is something that can be done elsewhere or manually. It's not really something you could test in code anyways.

Reply:
gotcha